pkg/dependencies/factory.go

```diff
@@ -659,7 +664,7 @@ func (f *Factory) WithHelm(opts ...executables.HelmOpt) *Factory {
 
 	f.buildSteps = append(f.buildSteps, func(ctx context.Context) error {
 		if f.registryMirror != nil {
-			opts = append(opts, executables.WithRegistryMirror(f.registryMirror.endpoint))
+			opts = append(opts, executables.WithRegistryMirror(f.registryMirror.endpoints[constants.DefaultRegistry]))
```

Comment:
we do this `endpoints[constants.DefaultRegistry]` in multiple places. It's probably worth to have another field in `f.registryMirror` for this

That said, shouldn't helm also support multiple registry mappings now? I think everywhere that used to understand the concept of registry mirror for public ecr should now understand the concept of multiple mappings